src/main/kotlin/com/seokjoo/todo/domain/service/trade/TodoTradeService.kt

```diff
@@ -0,0 +1,33 @@
+package com.seokjoo.todo.domain.service.trade
+
+import com.seokjoo.todo.common.exception.TodoException
+import com.seokjoo.todo.common.exception.TodoExceptionType
+import com.seokjoo.todo.domain.entity.todouser.User
+import com.seokjoo.todo.domain.service.auth.TodoAuthService
+import com.seokjoo.todo.domain.service.todo.TodoService
+import com.seokjoo.todo.domain.service.todo.TodoServiceResponseDTO
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+class TodoTradeService(
+    private val todoService: TodoService,
+    private val todoAuthService: TodoAuthService,
+) {
+    @Transactional
+    fun buyTodo(todoId: Long, user: User): TodoServiceResponseDTO {
+        // 1. 자기 todo 인지 확인
+        val todo = todoService.getTodoById(todoId)
+        if (todo.ownerId == user.userId) throw TodoException.of(TodoExceptionType.CAN_NOT_TRADE_OWN_TODO)
+        // 2. 아니라면 금액 충분한지 확인
+        user.money.isOutOfBalance(todo.price)
```

Comment:
```suggestion
        user.isAffordable(todo.price)
```

이건 어떠세요? 이렇게 하고, 여기서 `Money`까지 안 가고 `User`로만 예산 체크하면 좋을 것 같아요

Reply:
굿! 굳이 내부 요소를 뺄필욘없죠~